proposals/allow_time_based_window_for_policy_enforcement_behavior

```diff
@@ -0,0 +1,153 @@
+# Kyverno Design Proposal - Allow time-based window for policy enforcement behavior
+
+ISSUE : [#2233](https://github.com/kyverno/kyverno/issues/2233)
+
+Author: [Prateek Nandle](https://github.com/Prateeknandle)
+
+## Contents
+
+- [Kyverno Design Proposal - Allow time-based window for policy enforcement behavior](#kyverno-design-proposal---allow-time-based-window-for-policy-enforcement-behaviour)
+  * [Overview](#overview)
+  * [Proposal](#proposal)
+  * [Goals](#goals)
+  * [Example](#example)
+  * [Implementation](#implementation)
+    + [Change in Policy Behaviour]
+    + [Activity Status Report]
+
+## Overview
+
+I want to propose a new feature for policies, which will allow Time-based window policy enforcement behavior. This new feature would help to change the activity of the policies, for a particular defined Time window, eg : not allowing config changes during the weekends. Also a report will be created to see which policies are active at present.
+
+## Proposal 
+
+Currently, there is no such concept which will allow the user/ admin to make a time window where a policy’s activity would be changed automatically. Once a policy is defined then only manually it could be changed.
+
+Changing manually is a tedious job, for eg, if an admin wants that no developer should change config file during a time slot, let's say weekend, then the admin needs to manually change the policy. This is tedious, because every weekend the admin needs to change the policy manually.
+
+After this proposal the problem would be solved, a new tag would be added in the policy named, so that within a specified time window the policy would be inactive and outside the window it’ll be active.
+
+## Goals
+
+1. To add features for allowing users to define a time window where they want the particular policy to be inactive.
+
+2. To support users with a report for active/inactive status of policies at present time.
+
+3. To support the same feature for the CLI `test` command, i.e. functionality to add a time window to inactive a policy.
+
+## Example
+
+```
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: require-labels
+spec:
+  validationFailureAction: enforce
+  inactiveWindow: "00 17 * * * - 00 23 * * * "
```

Comment:
An initial idea was to allow the user to specify either `activeWindow` OR `inactiveWindow` depending on which is desired, but I think I agree it probably makes more sense to keep it simple and just go with `activeWindow`. I do think since this feature is a minority use case that it's best to start with a small, single enhancement to the schema (`spec.activeWindow`) and see where that gets us, then reevaluate in the future as further use cases are requested.

Reply:
done.